templates/CRM/UF/Form/Block.tpl

```diff
@@ -12,20 +12,37 @@
   {strip}
     {assign var=zeroField value="Initial Non Existent Fieldset"}
     {assign var=fieldset  value=$zeroField}
-    {include file="CRM/UF/Form/Fields.tpl"}
+    {* Unfortunately uF group information is munged into the uf fields array. We have ot iterate throug
+    to extract it. I n future we could migrate to a version of Block.tpl that expects the UFGroup
+    to be assigned by itself & remove this *}
+    {foreach from=$fields item=field key=fieldName}
+      {assign var=groupHelpPost  value=$field.groupHelpPost}
+      {assign var=groupHelpPre  value=$field.groupHelpPre}
+      {assign var=fieldset  value=$field.groupTitle}
+      {assign var=groupDisplayTitle value=$field.groupDisplayTitle}
+      {assign var=group_id value=$field.group_id}
+      {assign var=groupDisplayTitle value=$field.groupDisplayTitle}
```

Comment:
Minor thing is this is the same as line 22. Is there maybe a variable like groupTitle missing in this list? Except `fieldset` is groupTitle. This code is fun.

In terms of looping, I'm not sure it's worth the time but there is this: https://www.smarty.net/docsv2/en/language.function.foreach.tpl#foreach.property.first, i.e. add a `name=` to the foreach and then enclose the contents in an if block, such as:

```smarty
{foreach from=$fields item=field key=fieldName name=ufgroupfields}
  {if $smarty.foreach.ufgroupfields.first}
    ...
  {/if}
{/foreach} 
```

Reply:
@demeritcowboy I've removed the duplicate.

I'm not keen on switching to a smarty 2 function cos the risk of it being different when we finally upgrade smarty is higher...